File: BussinesLogic/Servicios/ServicioServicioLineaNegocio.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.ServicioLineaNegocio;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioServicioLineaNegocio : IServicioServicioLineaNegocio
    {
        public readonly IRepositorioServicioLineaNegocio repositorioServicioLineaNegocio;

        public ServicioServicioLineaNegocio(IRepositorioServicioLineaNegocio repositorioServicioLineaNegocio)
        {
            this.repositorioServicioLineaNegocio = repositorioServicioLineaNegocio;
        }
        public async Task<List<DtoServicioLineaNegocio>> ObtenerServicioLineaNegocio()
        {
            return await this.repositorioServicioLineaNegocio.ObtenerServicioLineaNegocio();
        }

        public async Task<DtoRespuestaSP> InsertarServicioLineaNegocio(EntitiServicioLineaNegocio entitiServicioLineaNegocio)
        {
            return await this.repositorioServicioLineaNegocio.InsertarServicioLineaNegocio(entitiServicioLineaNegocio);
        }

        public async Task<DtoRespuestaSP> ActualizarServicioLineaNegocio(EntitiServicioLineaNegocio entitiServicioLineaNegocio)
        {
            return await this.repositorioServicioLineaNegocio.ActualizarServicioLineaNegocio(entitiServicioLineaNegocio);
        }

        public async Task<DtoServicioLineaNegocio> ObtenerServicioLineaNegocioPorID(int idServicio)
        {
            return await this.repositorioServicioLineaNegocio.ObtenerServicioLineaNegocioPorID(idServicio);
        }
        public async Task<DtoRespuestaSP> EliminarServicioLineaNegocio(int idServicio)
        {
            return await this.repositorioServicioLineaNegocio.EliminarServicioLineaNegocio(idServicio);
        }
        public async Task<List<DtoLineaNegocio>> ObtenerLineaNegocioActivos()
        {
            return await this.repositorioServicioLineaNegocio.ObtenerLineasNegocioActivos();
        }
    }
}

File: CX/ClientApp/src/servicios/ServicioServicioLineaNegocio.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "serviciolineanegocio";

export const ObtenerServicioLineaNegocio = async () => {
    const url = `${controlador}/ObtenerServicioLineaNegocio`;
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerLineasNegociosActivos= async () => {
    const url = `${controlador}/ObtenerLineasNegociosActivos`;
    return await ProcesarDatosApi('GET', url);
}
export const AgregarServicioLineaNegocio = async (data) => {
    const url = `${controlador}/insertarserviciolineanegocio`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarServicioLineaNegocio = async (data) => {
    const url = `${controlador}/ActualizarServicioLineaNegocio`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarServicioLineaNegocio = async (id) => {
    const url = `${controlador}/eliminarserviciolineanegocio?idServicio=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}


export const ObtenerServicioLineaNegocioPorId = async (id) => {
    const url = `${controlador}/ObtenerServicioLineaNegocioPorID/${id}`;
    return await ProcesarDatosApi('GET', url);
}

